sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -209,47 +212,105 @@ class SmbMultichannel(GeneratedSmbMultichannel):
     enabled: Optional[bool]
     """If SMB Multichannel is enabled."""
 
-    def __init__(self, **kwargs: Any) -> None:
-        self.enabled = kwargs.get('enabled')
+    def __init__(self, *, enabled: Optional[bool] = None, **kwargs: Any) -> None:
+        self.enabled = enabled
         if self.enabled is None:
             raise ValueError("The value 'enabled' must be specified.")
 
 
+class SmbEncryptionInTransit(GeneratedSmbEncryptionInTransit):
+    """Settings for encryption in transit.
+
+    :keyword bool required: If encryption in transit is required.
+    """
+
+    required: Optional[bool]
+    """If encryption in transit is enabled."""
+
+    def __init__(self, *, required: bool, **kwargs: Any) -> None:
+        self.required = required
+
+
 class ShareSmbSettings(GeneratedShareSmbSettings):
     """Settings for the SMB protocol.
 
     :keyword SmbMultichannel multichannel: Sets the multichannel settings.
+    :keyword SmbEncryptionInTransit encryption_in_transit: Sets the encryption in transit settings.
     """
 
-    multichannel: SmbMultichannel
+    multichannel: Optional[SmbMultichannel]
```

Comment:
Technically this isn't `Optional` because the constructor will fail if `None` is provided?

Reply:
ooooo nevermind - are these mutually exclusive? If so - ignore me :)